src/pod/api.ts

```diff
@@ -1,27 +1,26 @@
 import { Bee, RequestOptions } from '@ethersphere/bee-js'
-import { EthAddress } from '@ethersphere/bee-js/dist/types/utils/eth'
 import { LookupAnswer } from '../feed/types'
 import { getFeedData } from '../feed/api'
 import { POD_TOPIC } from './personal-storage'
 import { ExtendedPodInfo, extractPods, PodsInfo } from './utils'
-import { prepareEthAddress } from '../utils/address'
-import { getWalletByIndex } from '../utils/wallet'
+import { getWalletByIndex, prepareEthAddress, privateKeyToBytes } from '../utils/wallet'
 import { List } from './list'
+import { utils } from 'ethers'
 
 /**
  * Gets pods list with lookup answer
  *
  * @param bee Bee instance
- * @param address Ethereum address
+ * @param userWallet root wallet for downloading and decrypting data
  * @param options request options
  */
-export async function getPodsList(bee: Bee, address: EthAddress, options?: RequestOptions): Promise<PodsInfo> {
+export async function getPodsList(bee: Bee, userWallet: utils.HDNode, options?: RequestOptions): Promise<PodsInfo> {
   let lookupAnswer: LookupAnswer | undefined
   let podsList = new List([], [])
 
   try {
-    lookupAnswer = await getFeedData(bee, POD_TOPIC, address, options)
-    podsList = extractPods(lookupAnswer.data.chunkContent())
+    lookupAnswer = await getFeedData(bee, POD_TOPIC, prepareEthAddress(userWallet.address), options)
+    podsList = extractPods(lookupAnswer.data.chunkContent(), privateKeyToBytes(userWallet.privateKey))
```

Comment:
after our discussion we can go with the symmetric encryption, since the symmetric encryption is more quantum computing resilient than the assymetric ones.